scikeras/wrappers.py

```diff
@@ -100,62 +162,73 @@ def __init__(
         self.validation_split = validation_split
         self.shuffle = shuffle
         self.run_eagerly = run_eagerly
+        self.epochs = epochs
 
         # Unpack kwargs
         vars(self).update(**kwargs)
 
+        # Check for deprecated APIs
+        if self.model is None and build_fn is not None:
+            self.model = build_fn
+            warnings.warn(
+                "`build_fn` will be renamed to `model` in a future release,"
+                " at which point use of `build_fn` will raise an Error instead."
+            )
+        for kwd in kwargs.keys():
+            if "__" not in kwd:
+                raise ValueError(
+                    "All kwargs must be routed parameters (ex: `model__hidden_layer_sizes`)"
+                    f" but {kwd} is not a routed parameter!"
+                )
+
     @property
     def __name__(self):
         return self.__class__.__name__
 
-    def _check_build_fn(self, build_fn):
-        """Checks `build_fn`.
+    @property
+    def _model_params(self):
+        return {
+            p.strip("model__")
```

Comment:
Careful. I'd use `p[len("model__"):]` instead of `p.strip("model__")`.

``` python
In [9]: s = "model__foo"

In [10]: s.strip("model__")
Out[10]: 'f'

In [11]: s[len("model__"):]
Out[11]: 'foo'
```

Strip works character-wise, so it strips out the characters "m", "o", "d", "e", "l" and "_".

I would expect the tests to catch this.

Reply:
This probably isn't the only place `strip` is being used improperly.

Reply:
Again, good catch. We should change it and test it.